core/src/themes/ionic.functions.string.scss

```diff
@@ -110,41 +94,74 @@
 
   @each $selector in $selectors {
     // If the selector contains :host( it means it is targeting a class on the host
-    // element so we need to change how we target it
+    // element so we need to change how we target it:
+    // @include add-root-selector(":host(.fixed)", "[dir=rtl]")
+    // --> [dir=rtl] :host(.fixed)
     @if str-contains($selector, ":host(") {
-      $shadow-element: str-replace($selector, ":host(", ":host-context(#{$addHostSelector}):host(");
-      $list: append($list, $shadow-element, comma);
-
       $new-element: ();
       $elements: str-split($selector, " ");
 
       @each $element in $elements {
         @if str-contains($element, ":host(") {
-          $scoped-element: $element;
+          // Replace the :host( and ) so all we have left is the class
+          // inside of it:
+          // :host(.fixed) -> .fixed
+          $updated-element: str-replace($element, ")", "");
+          $updated-element: str-replace($updated-element, ":host(", "");
+
+          // Add the class back inside of host with the rtl selector:
+          // .fixed -> [dir=rtl] :host(.fixed)
+          $updated-element: str-replace($updated-element, $updated-element, "#{$addHostSelector} :host(#{$updated-element})");
+
+          // Add the final selector after all transformations:
+          // :host(.fixed) -> [dir=rtl] :host(.fixed)
+          $new-element: append($new-element, $updated-element, space);
+        } @else {
+          // Add back any selectors that followed the host after transforming the
+          // first selector:
+          // :host(.fixed) ::slotted(ion-icon) -> [dir=rtl] :host(.fixed) ::slotted(ion-icon)
```

Comment:
I'm not sure I understand the output comment here. `[dir=rtl] :host` won't match anything: https://codepen.io/liamdebeasi/pen/ExdgJzg

Reply:
This outcome originated from [input outline](https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/input/input.md.outline.scss#L177).

Without `dir` in the front then it wouldn't compile the correct styling which is the following:

```css
[dir=rtl] :host(.input-fill-outline) .input-outline-start {
...
}
```

Reply:
So this only applies to non-shadow DOM components, correct?

Reply:
That is correct. The new changes would cause `add-root-selector` to not work with scoped or shadow DOM components. The main reason that `host-context` was removed is because when Firefox sees it then it completely removes it along with the attached list.

```css
:host(.popover-side-end) .popover-arrow, :host-contenxt(...)... {
...
}
```

Firefox wouldn't even apply the example above. It sees the `host-context` and ignores the entire style (this is the cause of the issue on item-sliding).

Another approach to solving the issue is to determine if the browser is Chrome then add `host-context` to the list. Thoughts?